### DISCUSSION TITLE
Creating a new Workspace next to the current Workspace, send the current node / App to and focus it?

### DISCUSSION THREAD

**Discussion by danscheer** [nikitabobko/AeroSpace#842] [questions-and-answers]
Hi, im using Yabai usually, since macos spaces are used it is possible to have empty Spaced i can send windows to, also sending it to the "next | prev" space. 

Since there are no empty Containers is Aerospace, it is not possible to say "Creating a new Workspace next to the current Workspace, send the current node / App to and focus it", right? 

The only way to use Aerospace is to create a specific Workspace (alt + 2) and send the App to it? 

Because i would like to not use dedicated keybinds for the Containers, i would like to have it more dynamicly. When i have focus on a window in worlspace 1, I want to send it to the next Workspace, basicly create one next to the current Workspace, and send the window to it. that way my setup would be created on the go, like a natural way. But "next | prev" only works when there already exists a Workspace (2). 

Any ideas?

**Comment by mpeterson** [comment]
Use [move-node-to-workspace](https://nikitabobko.github.io/AeroSpace/commands#move-node-to-workspace) with `--focus-follows-window`. When giving a workspace name as a parameter it will create the workspace if it doesn't exist. 

As workspaces are names, how do you define "next" will be up to you, but if you wanna script it you could use [list-workspaces](https://nikitabobko.github.io/AeroSpace/commands#list-workspaces), so "next" could be defined as `aerospace list-workspaces --count --all`

All in all, if you'd be triggering this from a script: 

```bash
aerospace move-node-to-workspace --focus-follows-window $(aerospace list-workspaces --count --all)
```

**Reply by danscheer** [reply]
Thanks for the Reply. 

Is it possible to make basic calculations? Because i need to adress the current Workspace -1 and +1 for this to work. 

`echo "$(($(aerospace list-workspaces --monitor focused --format %{workspace}) + 1))"`  

gives me the right output in the terminal, but the areospace.toml doesnt allow that. 

`'move-node-to-workspace `echo "$(($(aerospace list-workspaces --monitor focused --format %{workspace}) + 1))"` --focus-follows-window'`

was my try, and it threw an error when reloading the config.

**Comment by nikitabobko** [comment]
If we take the idea of `echo` subcommand from #278 + #930 then it will become possible to do so:

```toml
alt-right = 'echo "1 2 3 4 5 6 7 8 9 10" | aerospace move-node-to-workspace next --focus-follows-window'
```
